Tests/Packet++Test/Tests/TcpTests.cpp

```diff
@@ -122,17 +146,45 @@ PTF_TEST_CASE(TcpPacketWithOptionsParsing2)
 	PTF_ASSERT_EQUAL(sackPermOption.getValueAs<uint32_t>(), 0);
 	PTF_ASSERT_EQUAL(mssOption.getValueAs<uint32_t>(), 0);
 	PTF_ASSERT_EQUAL(mssOption.getValueAs<uint16_t>(1), 0);
+	//end deprecated
+
+	pcpp::TcpOption mssOption2 = tcpLayer->getTcpOption(pcpp::TcpOptionEnumType::Mss);
+	pcpp::TcpOption sackPermOption2 = tcpLayer->getTcpOption(pcpp::TcpOptionEnumType::SackPerm);
+	pcpp::TcpOption windowScaleOption2 = tcpLayer->getTcpOption(pcpp::TcpOptionEnumType::Window);
+	PTF_ASSERT_TRUE(mssOption2.isNotNull());
+	PTF_ASSERT_TRUE(sackPermOption2.isNotNull());
+	PTF_ASSERT_TRUE(windowScaleOption2.isNotNull());
+
+	PTF_ASSERT_EQUAL(mssOption2.getTcpOptionEnumType(), pcpp::TcpOptionEnumType::Mss, enumclass);
+	PTF_ASSERT_EQUAL(sackPermOption2.getTcpOptionEnumType(), pcpp::TcpOptionEnumType::SackPerm, enumclass);
+	PTF_ASSERT_EQUAL(windowScaleOption2.getTcpOptionEnumType(), pcpp::TcpOptionEnumType::Window, enumclass);
+
+	PTF_ASSERT_EQUAL(mssOption2.getTotalSize(), 4);
+	PTF_ASSERT_EQUAL(sackPermOption2.getTotalSize(), 2);
+	PTF_ASSERT_EQUAL(windowScaleOption2.getTotalSize(), 3);
+
+	PTF_ASSERT_EQUAL(mssOption2.getValueAs<uint16_t>(), htobe16(1460));
+	PTF_ASSERT_EQUAL(windowScaleOption2.getValueAs<uint8_t>(), 4);
+	PTF_ASSERT_EQUAL(sackPermOption2.getValueAs<uint32_t>(), 0);
+	PTF_ASSERT_EQUAL(mssOption2.getValueAs<uint32_t>(), 0);
+	PTF_ASSERT_EQUAL(mssOption2.getValueAs<uint16_t>(1), 0);
 
 	pcpp::TcpOption curOpt = tcpLayer->getFirstTcpOption();
+	//TODO: remove deprecated
```

Comment:
Maybe remove this comment? It's a bit confusing because it hints removing the entire code until the end of the test, which is incorrect. Another option is to split into deprecated and new tests like in the previous tests:

```cpp
//TODO: remove deprecated
pcpp::TcpOption curOpt = tcpLayer->getFirstTcpOption();
PTF_ASSERT_TRUE(curOpt.isNotNull() && curOpt.getTcpOptionType() == pcpp::TCPOPT_MSS);
curOpt = tcpLayer->getNextTcpOption(curOpt);
PTF_ASSERT_TRUE(curOpt.isNotNull() && curOpt.getTcpOptionType() == pcpp::TCPOPT_SACK_PERM);
...
//end deprecated

curOpt = tcpLayer->getFirstTcpOption();
PTF_ASSERT_TRUE(curOpt.isNotNull() && curOpt.getTcpOptionEnumType() == pcpp::TcpOptionEnumType::Mss);
curOpt = tcpLayer->getNextTcpOption(curOpt);
PTF_ASSERT_TRUE(curOpt.isNotNull() && curOpt.getTcpOptionEnumType() == pcpp::TcpOptionEnumType::SackPerm);
...
```